example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,370 @@
+    if (_focalPoint != value) {
+      _focalPoint = value;
+      markNeedsLayout();
+    }
+  }
+
+  EdgeInsets? _padding;
+  EdgeInsets? get padding => _padding;
+  set padding(EdgeInsets? value) {
+    if (_padding != value) {
+      _padding = value;
+      markNeedsLayout();
+    }
+  }
+
+  Color _backgroundColor;
+  Color get backgroundColor => _backgroundColor;
+  set backgroundColor(Color value) {
+    if (value != _backgroundColor) {
+      _backgroundColor = value;
+      _backgroundPaint = Paint()..color = _backgroundColor;
+      markNeedsPaint();
+    }
+  }
+
+  bool _allowHorizontalArrow;
+  bool get allowHorizontalArrow => _allowHorizontalArrow;
+  set allowHorizontalArrow(bool value) {
+    if (value != _allowHorizontalArrow) {
+      _allowHorizontalArrow = value;
+      markNeedsLayout();
+    }
+  }
+
+  late Paint _backgroundPaint;
+
+  @override
+  void performLayout() {
+    // We need to know our offset in order to calculate the arrow direction
+    // and we don't know our offset until the paint phase.
+    // Therefore, we need to reserve space for the arrow in both axes.
+    final reservedSize = Size(
+      (padding?.horizontal ?? 0) + arrowLength * 2,
+      (padding?.vertical ?? 0) + arrowLength * 2,
+    );
+
+    // Compute the child constraints to leave space for the arrow.
+    final innerConstraints = constraints.enforce(
+      BoxConstraints(
+        maxHeight: constraints.maxHeight - reservedSize.height,
+        maxWidth: constraints.maxWidth - reservedSize.width,
+      ),
+    );
+
+    child!.layout(innerConstraints, parentUsesSize: true);
+
+    size = constraints.constrain(Size(
+      reservedSize.width + child!.size.width,
+      reservedSize.height + child!.size.height,
+    ));
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    final localFocalPoint = globalToLocal(focalPoint);
+
+    final direction = _computeArrowDirection(offset & size, focalPoint);
+    final arrowCenter = _computeArrowCenter(direction, localFocalPoint);
+    final contentOffset = _computeContentOffset(direction, arrowLength);
+
+    final borderPath = _buildBorderPath(direction, arrowCenter);
+
+    context.canvas.drawPath(borderPath.shift(offset), _backgroundPaint);
+
+    if (child != null) {
+      context.paintChild(child!, offset + contentOffset);
+    }
+  }
+
+  @override
+  bool hitTest(BoxHitTestResult result, {required Offset position}) {
+    if (hitTestChildren(result, position: position)) {
+      return true;
+    }
+    // Allow hit-testing around the content, e.g, we might have padding and
+    // the user is trying to drag using the padding area.
+    final rect = Offset.zero & size;
+    return rect.contains(position);
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
+    final direction = _computeArrowDirection(Offset.zero & size, focalPoint);
+    final contentOffset = _computeContentOffset(direction, arrowLength);
```

Comment:
How does the arrow direction change the content offset?

Reply:
It doesn't affect it anymore. I updated it now.